client/src/actions/index.js

```diff
@@ -43,7 +43,7 @@ async function configFetch(dispatch) {
 
 async function userInfoFetch(dispatch) {
   return fetchJson("userinfo").then((response) => {
-    const userinfo = { ...response.userinfo };
+    const userinfo = { ...(response ? response.userinfo : null) };
```

Comment:
We can simplify this to a conditional destructure like this:
```suggestion
    const { userinfo } = response || {};
```

Reply:
(The line should have been: `const { userinfo } = response` to begin with.)

Reply:
Thanks, fixed.